src/application.rs

```diff
@@ -1,70 +1,77 @@
 use std::collections::BTreeSet;
-use std::error::Error;
+
+use rocket::fairing::AdHoc;
+use serde::Deserialize;
 
 use super::routes;
 use super::storage::{SqlStorage, Storage};
-use super::web::{AuthValidator, JwtValidator, RequestIdHeader, RequestSpan};
+use super::web::{AuthValidator, JwtValidator, RequestIdHeader};
 
-#[derive(Debug)]
+#[derive(Debug, Deserialize)]
 pub struct Config {
+    /// Database connection string
+    pub database_url: String,
+
     /// Keeps a set of supported syntaxes. When set, RESTful API rejects
     /// snippets with *unsupported* syntaxes. Normally this set must be
     /// kept in sync with a set of supported syntaxes in XSnippet Web in
     /// order to ensure that the web part can properly syntax-highlight
     /// snippets.
+    #[serde(default)]
     pub syntaxes: Option<BTreeSet<String>>,
 
     /// The intended recipient of the tokens (e.g. "https://api.xsnippet.org")
+    #[serde(default = "default_jwt_audience")]
     pub jwt_audience: String,
     /// The principal that issues the tokens (e.g. "https://xsnippet.eu.auth0.com/")
+    #[serde(default = "default_jwt_issuer")]
```

Comment:
Makes me wonder, is this even a good idea to have _defaults_ for JWT settings? Especially the one tied to Auth0? :thinking:

Reply:
Good question! I don't remember our original thinking about this... On the one hand, I do like that we need to specify fewer config options and use "production defaults". But I also see your point about tying this to Auth0. I'll keep like this for the purpose of migration to a new Rocket release. Let's figure out what we want to do!